src/tess.py

```diff
@@ -0,0 +1,364 @@
+                    df = df[~bad_idx]
+
+                if len(df) > 0:
+                    dfs.append(df)
+
+            prev_idx = idx + 1
+
+        new_datasets = []
+        keep_columns = [
+            time_col,
+            exp_col,
+            filt_col,
+            mag_col,
+            magerr_col,
+            ra_col,
+            dec_col,
+            flag_col,
+        ]
+        for df in dfs:
+            df = df[keep_columns]
+            if len(df) > 0:
+                new_datasets.append(Lightcurve(data=df, **init_kwargs))
+
+        return new_datasets
+
+    def fetch_data_from_observatory(self, cat_row, verbose=0, **kwargs):
+        """
+        Fetch data from TESS for a given source.
+        Must return a dataframe (or equivalent), even if it is an empty one.
+        This must be implemented by each observatory subclass.
+
+        Parameters
+        ----------
+        cat_row: dict like
+            A row in the catalog for a specific source.
+            In general, this row should contain the following keys:
+            name, ra, dec, mag, filter_name (saying which band the mag is in).
+        kwargs: dict
+            Additional keyword arguments to pass to the fetcher.
+
+        Returns
+        -------
+        data : pandas.DataFrame or other data structure
+            Raw data from the observatory, to be put into a RawData object.
+        altdata: dict
+            Additional data to be stored in the RawData object.
+
+        """
+        # TODO: figure out dstArcSec threshold for flagging close stars
+        DST_THRESHOLD = 10
+        # TODO: figure out mag diff threshold for flagging similar stars
+        MAGDIFF_THRESHOLD = 0.5
+        # TODO: figure out optimal radii for queries
+        
+        # name should be source_id for GAIA catalog entry
+        name = cat_row['name']
+        dec = cat_row['dec']
+        ra = cat_row['ra']
+        mag = cat_row['mag']
+        if mag > 16:
+            # TESS can't see stars fainter than 15 mag
+            print(f"Magnitude of {mag} is too faint for TESS.")
+            return pd.DataFrame()
+
+        cone_radius = kwargs['cone_radius'] \
+                        if 'cone_radius' in kwargs else 0.01
+        sign = ' -' if dec < 0 else ' '
+        coord_str = str(ra) + sign + str(dec)
+        
+        kwargs = {
+            'coordinates': coord_str,
+            'catalog': 'TIC',
+            'radius': cone_radius
+        }
+        catalog_data = self._try_query(Catalogs.query_region, kwargs)
+        if len(catalog_data) == 0:
+            print("No TESS object found for given catalog row.")
+            return pd.DataFrame()
+        
+        # TODO: check for better way of selecting candidate
+        # catalog_data is automatically sorted by dstArcSec, index 0 is closest
+        # also, is GAIAmag guaranteed to exist? 
+        # what if we're not using a GAIA cat_row?
```

Comment:
I think the function you are using gets matches from Gaia regardless of the input you give it (assuming you give it `'catalog': 'TIC'`). 
So even if we don't use Gaia to make the `cat_row`, it shouldn't matter. All you really input to the TESS tools are the RA/Dec and radius.

Reply:
There could be missing Gaia magnitude (probably it would be NaN) but we should think what is the best action in that case. Maybe just assume a mag=NaN is out of range no matter what?

Reply:
For now I think it makes sense to ignore NaN values of GAIAmag.
I think my second question should be rephrased as: which mag do we compare by if cat_row is from some non-GAIA catalog? Currently it's hard coded to compare by GAIAmag, but there are many other mag options like B, V, u, g, r, i, etc.

Reply:
Yeah so this part goes in the `Catalog` class. There's a parameter called `mag_column` that tells the catalog which column of the raw data table it should pick up as "the magnitude". It is true that we can probably do better than this, e.g., pick up multiple magnitudes and use the Gaia G for matching to ZTF g and Gaia RP to match to ZTF r, but that would complicate things a lot. That's why I'm going to use this magnitude only as a very rough estimate for what the source should be. Potentially if the magnitude from the catalog is in G, and the lightcurves are in R, it could make a big difference if the source has strong colors. So this is another reason to be fairly loose on the magnitude limits.